src/ui/include/EditorNS/editor.h

```diff
@@ -267,29 +431,99 @@ namespace EditorNS
          */
         void setCodec(QTextCodec *codec);
 
+        /**
+         * @brief Whether or not the editor is using a byte order mark
+         * @return bool
+         */
         bool bom() const;
+        /**
+         * @brief Enable/disable byte order mark for the editor.
+         * @param bom
+         */
         void setBom(bool bom);
 
+        /**
+         * @brief Returns a list of themes available to the editor.
+         * @return QList<Theme>
+         */
         QList<Theme> themes();
+        /**
+         * @brief Sets the theme for the editor.
+         * @param theme
+         */
         void setTheme(Theme theme);
+        /**
+         * @brief Returns a Theme struct for a given name.  An emptry struct
+         *        is returned if no theme is found.
+         * @param name
+         * @return Theme struct
+         */
         static Editor::Theme themeFromName(QString name);
  
-
-        void setSelectionsText(const QStringList &texts, selectMode mode);
+        /**
+         * @brief Sets the selected text within the editor.
+         * @param texts
+         * @param selectMode
+         */
+        void setSelectionsText(const QStringList &texts, SelectMode mode);
         void setSelectionsText(const QStringList &texts);
+
+        /**
+         * @brief Sets the selection range for the editor.
+         * @param fromLine
+         * @param fromCol
+         * @param toLine
+         * @param toCol
+         */
         void setSelection(int fromLine, int fromCol, int toLine, int toCol); 
 
+        /**
+         * @brief Enable/disable overwrite mode within the editor.
+         * @param overwrite.
+         */
         void setOverwrite(bool overwrite);
+
+        /**
+         * @brief Set the visibility of tab characters within the editor.
+         * @param visible
+         */
         void setTabsVisible(bool visible);
 
         /**
-         * @brief Detect the indentation mode used within the current document.
-         * @return
+         * @brief Get the indentation mode for the current document.
+         * @return Editor::IndentationMode struct
          */
-        Editor::IndentationMode detectDocumentIndentation(bool *found = nullptr);
         Editor::IndentationMode indentationMode();
 
+        /**
+         * @brief Set the indentation mode for the current document.
+         * If transport is set to false, the data isn't sent to javascript.
+         * @param useTabs
+         * @param size
+         * @param custom
+         * @param transport
+         */
+        void setIndentationMode(const bool useTabs, const int size, 
```

Comment:
It was to make it a multi-purpose function where we could initialize setting without needing a secondary function... though I could probably just write an initDefaults type function for this purpose for when it needs to be extended in the future.